truss/contexts/local_loader/truss_file_syncer.py

```diff
@@ -1,27 +1,82 @@
+import logging
 from pathlib import Path
 from threading import Thread
+from typing import Optional
 
-from truss.util.path import copy_file_path
+from truss.constants import CONFIG_FILE
+from truss.patch.calc_patch import calc_config_patches
+from truss.patch.local_truss_patch_applier import LocalTrussPatchApplier
+from truss.templates.control.control.helpers.types import (
+    Action,
+    ModelCodePatch,
+    Patch,
+    PatchType,
+)
+from truss.truss_config import TrussConfig
+from truss.truss_spec import TrussSpec
+from truss.util.path import file_content, is_ignored, load_trussignore_patterns
 from watchfiles import Change, watch
 
+OP_2_ACTION = {
+    Change.added: Action.ADD,
+    Change.deleted: Action.REMOVE,
+    Change.modified: Action.UPDATE,
+}
+
+
+class TrussPatchEmitter:
+    def __init__(
+        self,
+        truss_dir: Path,
+        logger: logging.Logger,
+    ) -> None:
+        self._truss_dir = truss_dir
+        self._config = TrussConfig.from_yaml(self._truss_dir / CONFIG_FILE)
+        self._logger = logger
+
+    def __call__(self, op, path: Path) -> Optional[Patch]:
+        truss_spec = TrussSpec(self._truss_dir)
+        model_module_path = str(
+            truss_spec.model_module_dir.relative_to(self._truss_dir)
+        )
+        if str(path).startswith(model_module_path):
+            rel_path = str(path.relative_to(model_module_path))
+            return Patch(
+                type=PatchType.MODEL_CODE,
+                body=ModelCodePatch(
+                    OP_2_ACTION[op],
+                    rel_path,
+                    file_content(self._truss_dir / path),
+                ),
+            )
+        if str(path) == CONFIG_FILE:
+            new_config = TrussConfig.from_yaml(self._truss_dir / CONFIG_FILE)
+            config_patches = calc_config_patches(self._config, new_config)
+            self._config = new_config
+            return config_patches[0] if config_patches else None
```

Comment:
why only return once? It's working making this interface return a list to make sure that we're not dropping anything that can be patch. 
Image if the user updated both system_packages and requirements before saving. we'd be dropping one of them here

Reply:
We emit patches on a per `Change` basis from `TrussFileSyncer.run` – so there should only ever be one patch generated

Reply:
That's not necessary true. The Change's emitted from watchile correspond to files. Each file change can result in multiple patches. The example i wrote above about system_packages and requirements is one such case.
There is no break in interface by mapping a `Change` to `List[Patch]`. The point of this class is to do that conversion in a way most compatible with the control server architecture, as opposed to what's natural for the underlying library we're using.

Reply:
Ah, yeah you are right I didn't read your comment carefully with the counterexample – let me change the interface to `List[Patch]`